crypto/keyring/keyring.go

```diff
@@ -201,7 +201,7 @@ type keystore struct {
 	options Options
 }
 
-func newKeystore(kr keyring.Keyring, cdc codec.Codec, backend string, opts ...Option) keystore {
+func NewKeystore(kr keyring.Keyring, cdc codec.Codec, backend string, opts ...Option) keystore {
```

Comment:
We have a public API that returns a private type -- isn't that a codesmell?

Reply:
<img width="534" alt="image" src="https://user-images.githubusercontent.com/20993683/180174081-7e478aaf-bc8a-4275-83b0-99c8b4c80cb3.png">

I think it is not a problem, we can do it like this in the client

Reply:
Currently I use file, I want the system to enter the password automatically when starting (CI,CD). Currently I can't do it because `newKeystore` is private. I think it should be public.

Reply:
My password is saved in vault

Reply:
I wonder if we should make `keystore` public then. WDYT @amaurym ?

Reply:
`keyring.New` has a password input of `userInput io.Reader`. How do I write password entry automation?
Currently I'm needing to manually enter the password every time I boot. I need to self-enter at CI/CD

Reply:
You pipe to STDIN

Reply:
> We have a public API that returns a private type -- isn't that a codesmell?

Yes. Exported methods/functions should never return unexported types.

Reply:
How to use `STDIN` to autofill password ?
Hmm if so, I think I'll try creating a fun like this:
```
func newFileBackendKeyringWithPassword(password string, opts ...Option) Keyring {}
```
Is it ok?

Reply:
That doesn't seem safe either as you don't want to keep a passphrase in memory. To pipe into STDIN:

```shell
some_command <<< $PASSPHRASE
```